src/box/applier.cc

```diff
@@ -2296,6 +2296,7 @@ applier_thread_data_destroy(struct applier *applier)
 		say_warn("applier data destruction is continued");
 	});
 
+	diag_clear(&applier->thread.exit_msg.diag);
```

Comment:
Let's move it to `applier_thread_detach_applier` for the sake of symmetry.

Reply:
Fixed.

```diff
--- a/src/box/applier.cc
+++ b/src/box/applier.cc
@@ -2269,6 +2269,7 @@ applier_thread_detach_applier(struct cbus_call_msg *base)
        lsregion_destroy(&applier->thread.lsr);
        fiber_cond_destroy(&applier->thread.writer_cond);
        ibuf_destroy(&applier->thread.ibuf);
+       diag_clear(&applier->thread.exit_msg.diag);

        return 0;
 }
@@ -2296,7 +2297,6 @@ applier_thread_data_destroy(struct applier *applier)
                say_warn("applier data destruction is continued");
        });

-       diag_clear(&applier->thread.exit_msg.diag);
        fiber_cond_destroy(&applier->msg_cond);
 }
```